alg_ed25519/alg.go

```diff
@@ -0,0 +1,53 @@
+package alg_ed25519
+
+import (
+	"context"
+	"crypto/ed25519"
+	"errors"
+
+	"github.com/common-fate/httpsig"
+	"github.com/common-fate/httpsig/contentdigest"
+	"github.com/common-fate/httpsig/signer"
+	"github.com/common-fate/httpsig/verifier"
+)
+
+const Ed25519Alg = "ed25519"
+
+type Ed25519 struct {
+	PrivateKey ed25519.PrivateKey
+	PublicKey  ed25519.PublicKey
+	Attrs      any
+}
+
+var _ verifier.Algorithm = &Ed25519{}
+var _ signer.Algorithm = &Ed25519{}
+var _ httpsig.Attributer = &Ed25519{}
+
+func (a Ed25519) Attributes() any {
+	return a.Attrs
+}
+
+func (a Ed25519) Type() string {
+	return Ed25519Alg
+}
+
+func (a Ed25519) ContentDigest() contentdigest.Digester {
+	return contentdigest.SHA512
```

Comment:
I just picked sha512, since its independent of the actual signing algorithm. Let me know if you want something different

Reply:
Looks good, this could be made customisable in future too with a field on the `Ed25519` struct. It seems like in this instance, coupling the `ContentDigest` to the `Ed25519` is a poor abstraction because we force you to specify it in the algorithm.

Reply:
Yea, it might be nice to decouple those